File: Eye-openingJava/src/main/java/com/example/eye_openingJava/util/TypeFaceUtil.java
package com.example.eye_openingJava.util;

import android.content.Context;
import android.graphics.Typeface;

public class TypeFaceUtil {

    public static final int FZLL_TYPEFACE = 1;
    public static final int FZDB1_TYPEFACE = 2;
    public static final int FUTURA_TYPEFACE = 3;
    public static final int DIN_TYPEFACE = 4;
    public static final int LOBSTER_TYPEFACE = 5;

    public static Typeface getFontTypeface(Context context, int typeface) {
        switch (typeface) {
            case FZLL_TYPEFACE:
                return Typeface.createFromAsset(context.getAssets(), "fonts/FZLanTingHeiS-L-GB-Regular.TTF");
            case FZDB1_TYPEFACE:
                return Typeface.createFromAsset(context.getAssets(), "fonts/FZLanTingHeiS-DB1-GB-Regular.TTF");
            case FUTURA_TYPEFACE:
                return Typeface.createFromAsset(context.getAssets(), "fonts/Futura-CondensedMedium.ttf");
            case DIN_TYPEFACE:
                return Typeface.createFromAsset(context.getAssets(), "fonts/DIN-Condensed-Bold.ttf");
            case LOBSTER_TYPEFACE:
                return Typeface.createFromAsset(context.getAssets(), "fonts/Lobster-1.4.otf");
            default:
                return Typeface.DEFAULT;
        }
    }
}

File: Eye-openingKotlin/src/main/java/com/example/eye_openingKotlin/util/TypeFaceUtil.kt
package com.example.eye_openingKotlin.util

import android.graphics.Typeface
import com.example.eye_openingKotlin.EyeopeningApplication

/**
 * 自定义字体工具类。
 *
 * @author vipyinzhiwei
 * @since  2020/5/24
 */
object TypeFaceUtil {

    const val FZLL_TYPEFACE = 1

    const val FZDB1_TYPEFACE = 2

    const val FUTURA_TYPEFACE = 3

    const val DIN_TYPEFACE = 4

    const val LOBSTER_TYPEFACE = 5

    val fzlLTypeface by lazy {
        Typeface.createFromAsset(EyeopeningApplication.context.assets, "fonts/FZLanTingHeiS-L-GB-Regular.TTF")
    }

    val fzdb1Typeface by lazy {
        Typeface.createFromAsset(EyeopeningApplication.context.assets, "fonts/FZLanTingHeiS-DB1-GB-Regular.TTF")
    }

    val futuraTypeface by lazy {
        Typeface.createFromAsset(EyeopeningApplication.context.assets, "fonts/Futura-CondensedMedium.ttf")
    }

    val dinTypeface by lazy {
        Typeface.createFromAsset(EyeopeningApplication.context.assets, "fonts/DIN-Condensed-Bold.ttf")
    }

    val lobsterTypeface by lazy {
        Typeface.createFromAsset(EyeopeningApplication.context.assets, "fonts/Lobster-1.4.otf")
    }
}